projects/igniteui-angular/src/lib/grids/tree-grid/tree-grid-selection.service.ts

```diff
@@ -0,0 +1,280 @@
+import { Injectable } from '@angular/core';
+import { IgxGridSelectionService } from '../selection/selection.service';
+import { ITreeGridRecord } from './tree-grid.interfaces';
+
+
+@Injectable()
+export class IgxTreeGridSelectionService extends IgxGridSelectionService {
+    private rowsToBeSelected: Set<any>;
+    private rowsToBeIndeterminate: Set<any>;
+
+    /** Select specified rows. No event is emitted. */
+    selectRowsWithNoEvent(rowIDs: any[], clearPrevSelection?): void {
+        if (this.grid && this.grid.rowSelection === 'multipleCascade') {
+            this.cascadeSelectRowsWithNoEvent(rowIDs, clearPrevSelection);
+            return;
+        }
+        super.selectRowsWithNoEvent(rowIDs, clearPrevSelection);
+    }
+
+    /** Deselect specified rows. No event is emitted. */
+    deselectRowsWithNoEvent(rowIDs: any[]): void {
+        if (this.grid.rowSelection === 'multipleCascade') {
```

Comment:
Checks should be performed against the enum, rather than against the string representation. This way we can change the string without having to find and replace it everywhere!